mutt_notmuch.c

```diff
@@ -1792,23 +1799,22 @@ bool nm_normalize_uri(char *new_uri, const char *orig_uri, size_t new_uri_sz)
   struct Context tmp_ctx;
   struct NmCtxData tmp_ctxdata;
 
-  tmp_ctx.magic = MUTT_NOTMUCH;
-  tmp_ctx.data = &tmp_ctxdata;
-  tmp_ctxdata.db_query = NULL;
+  struct NmCtxData *tmp_data_from_uri = new_ctxdata(strdup(orig_uri));
```

Comment:
I can't see the need for this `strdup()`
Inside `new_ctxdata()` there are calls to `mutt_substrdup()` to keep the bits we want to keep.

Reply:
Without the `strdup()` I receive a discarded qualifier warning during compilation. Would casting it to `char *` be a better alternative?

Reply:
Not a good one :) How about changing `new_ctxdata` to take a `const char*`?